programs/mpl-core/src/plugins/lifecycle.rs

```diff
@@ -342,15 +342,28 @@ impl Plugin {
         plugin: &Plugin,
         ctx: &PluginValidationContext,
     ) -> Result<ValidationResult, ProgramError> {
+        let target_plugin = ctx.target_plugin.ok_or(MplCoreError::InvalidPlugin)?;
+
         // If the plugin being checked is Authority::None then it can't be revoked.
         if ctx.self_authority == &Authority::None
-            && ctx.target_plugin.is_some()
-            && PluginType::from(ctx.target_plugin.unwrap()) == PluginType::from(plugin)
+            && PluginType::from(target_plugin) == PluginType::from(plugin)
         {
             return reject!();
         }
 
-        match plugin {
+        let base_result = if ctx.self_authority
+            == &(Authority::Address {
+                address: *ctx.authority_info.key,
+            })
```

Comment:
I think there may be an issue with this, which isn't related to your PR but a problem with this logic in general.

If you set the authority to the Update Authority on an owner-managed plugin, and then you sign revokePluginAuthority as the Update Authority, it won't be approved by the asset, and since it is not a plain address, it will not be approved here by the plugin, so the Update Authority would never be able to self-revoke.

I think its a similar problem for an authority-managed plugin with authority set to Owner. In that case, Owner would never be able to self-revoke.